src/next/components/Snippet/UsercentricsSmartDataProtector.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import React, {ReactElement} from "react";
+import Script from "next/script";
+
+type WindowWithUsercentricsSmartDataProtector = Window & Readonly<{
+	uc?: {
+		blockElements(elements: Record<string, string>): void;
+		reloadOnOptIn(key: string): void;
+		reloadOnOptOut(key: string): void;
+	};
+}>;
+
+export const USERCENTRICS_SERVICE_PROVIDER_KEYS = {
+	GOOGLE_MAPS: "S1pcEj_jZX",
+} as const;
+
+type UsercentricsSerivceProviderInternalKeys = keyof typeof USERCENTRICS_SERVICE_PROVIDER_KEYS;
+type UsercentricsSerivceProviderKey = typeof USERCENTRICS_SERVICE_PROVIDER_KEYS[UsercentricsSerivceProviderInternalKeys] | string;
```

Comment:
We should prefer using `… | (string & {})` to get proper auto-completion and allow unknown service keys as well. See https://twitter.com/mattpocockuk/status/1823380970147369171 for a demonstration on what it does.

Reply:
With the current definition, autocompletion is actually already possible.